### PR TITLE
Export final final actual revision

### PR DESCRIPTION
**Pull request by guzzim2022**
Pulling all the exporting files from my branch to the main. Contains the latest updates on the exporting test. Contains the SOLID and clean latest version of the exporting feature. The confirmation screen is added and works. Orders were removed and will not be exported as they are not considered to be part of the inventory yet. The exporter class does not violate the clean architecture principle by directly accessing entities, now it has a controller, an interactor, and two interfaces to connect the layers.

### CODE REVIEW

File: src/main/java/Screens/ExportSuccessfulUI.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 public class ExportSuccessfulUI {
     /**
```
Comment (chenbriann) [inline src/main/java/Screens/ExportSuccessfulUI.java:7]: Try to make UI popups cleaner and more user friendly in the future.

File: src/test/java/entities/ExporterTest.java [hunk 2]
```diff
@@ -148,20 +149,20 @@ public void InventoryViewModel() {
...
+//    @Test
+//    public void ExporterInventory() throws IOException {
+//        ExportPresenter presenter = new ExportInventoryUpdater();
+//        HashMap<String, Object> controllers = new HashMap<>();
+//        ExporterInventory exporterInventory = new ExporterInventory(inventory,presenter, controllers);
+//        ExportRequestModel exportRequestModel = new ExportRequestModel(inventoryname, inventoryTable);
+//        exporterInventory.create(exportRequestModel);
+//
+//        Assertions.assertEquals(inventoryTable[0][0], "bananas");
+//    }
```
Comment (chenbriann) [inline src/test/java/entities/ExporterTest.java:166]: You should also create a test for your export use case interactor to test if it correctly turns the inventory data into data that is ready to export.

File: src/test/java/entities/ExporterTest.java [hunk 3]
```diff
@@ -91,7 +92,7 @@ public ExporterTest() {
 
     /**
      * Asserts that the two files of the exported inventory exists at the specificed location
-     * @throws IOException as required by Exporter
+     * @throws IOException as required by Exporter class
      */
 
     @Test
```
Comment (chenbriann) [inline src/test/java/entities/ExporterTest.java:98]: Is it possible to have the user specify the export path instead of hard coding it in?

**Review by chenbriann — APPROVED** [review]
Export feature looks good. Good work with the addition of clean architecture to your code, as it was lacking in your last pull request. Wrapping up inventory data in a wrapper was a good way to separate the entity layer and the database layer. In the future, consider adding interactor tests that implement an anonymous presenter to test if your interactor correctly changes data. Also, it would be useful if the user could specify the export file path instead of always setting it to the exports folder.

**Review by DaneshKohina — APPROVED** [review]
Well done Matteo this feature has come a long way. There was no clean architecture last time but in this pull request you have implemented it really well. Good use of the input boundary and data wrapper as this separates the enterprise business rules layer and the frameworks and drivers layers. I think there is a better way to store the exported files rather than just putting it in a exports folder. You did a great job following SOLID design principles too. Maybe next tine you can implement a design pattern.

**Review by BENEDIDIN — APPROVED** [review]
Matteo, the work you have done is truly magnificent. Clean architecture has been implemented throughout the program, ensuring our grade in this project will be satisfactory. The grammatical fixes you have made to your comments will greatly contribute to the ease of understanding. This exporting feature will ensure our inventories will be able to live much longer lives, so for that reason this is an excellent addition to our code.

**Pull request merged**